### PR TITLE
Rename `EditionId` to `RecordId`

### PR DESCRIPTION
**Pull request by Alfred-Mountfield**
## 🌟 What is the purpose of this PR?

As part of bringing HASH back in sync with the changes made in the Block Protocol for the new specification, we need to rename various fields. This PR renames `EditionId` to `RecordId` within the TypeScript code, and introduces some temporary mapping functions which should allow us to merge at least a few PRs without needing a dev branch.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1203358502199087/1203756383156843/f) _(internal)_

## 🚫 Blocked by

- [x] #2040 

## 🔍 What does this change?

- Renames `EntityEditionId` to `EntityRecordId`
  - Renames the fields inside it to be consistent with the `0.3` branch of the BP
- Renames `OntologyTypeEditionId` to `OntologyTypeRecordId`
  - Renames the fields inside it to be consistent
- Introduces correct `VertexId` types
- Creates temporary mapping functions.   

## 📜 Does this require a change to the docs?

N/A

## ⚠️ Known issues

Many more inconsistencies to be addressed before we can remove the mapping functions

## 🐾 Next steps

- Rename `EditionId` in the Rust Graph Service
- Continue migrating types

## 🛡 What tests cover this?

All the normal ones

## ❓ How to test this?

Test as usual, ensure the app works and CI passes

## 📹 Demo

N/A

### CODE REVIEW

**Comment by Alfred-Mountfield** [comment]
@nonparibus you seem to have been tagged because this was stacked on the prior PR you approved (even though this no longer has changes for a license file. Kind of frustrating code owners isn't removing it

**Review by nonparibus — APPROVED** [review]

**Review by thehabbos007 — DISMISSED** [review]

**Review by nonparibus — DISMISSED** [review]

**Review by nonparibus — APPROVED** [review]

**Review by nonparibus — DISMISSED** [review]

File: libs/@local/hash-subgraph/src/types/element.ts [hunk 1]
```diff
@@ -5,41 +5,48 @@ import {
...
-  DataTypeWithMetadataGraphApi,
-  "schema"
-> & { schema: DataType };
+export type OwnedOntologyElementMetadata = {
+  recordId: OntologyTypeRecordId;
+  ownedById: string;
+  provenance: ProvenanceMetadataGraphApi;
+};
+
+export type ExternalOntologyElementMetadata = {
```
Comment (TimDiekmann) [inline libs/@local/hash-subgraph/src/types/element.ts:27]: We also have `provenance` here to track who triggered that record creation in the database

**Review by thehabbos007 — DISMISSED** [review]

**Review by nonparibus — APPROVED** [review]

File: libs/@local/hash-subgraph/src/types/element.ts (hunk 1 above)
Comment (Alfred-Mountfield) [reply libs/@local/hash-subgraph/src/types/element.ts:27]: Have added, thank you

**Review by thehabbos007 — APPROVED** [review]

**Pull request merged**